mypy/checkexpr.py

```diff
@@ -3411,6 +3411,10 @@ def visit_yield_from_expr(self, e: YieldFromExpr, allow_none_return: bool = Fals
         elif self.chk.type_is_iterable(subexpr_type):
             if is_async_def(subexpr_type) and not has_coroutine_decorator(return_type):
                 self.chk.msg.yield_from_invalid_operand_type(subexpr_type, e)
+            elif isinstance(subexpr_type, TupleType):
+                # Tuples get special handling here, since they "don't have __iter__"...
+                item_type = join.join_type_list(subexpr_type.items)
+                subexpr_type = self.chk.named_generic_type('typing.Iterable', [item_type])
```

Comment:
Again, why this is needed? Why `check_method_call_by_name()` doesn't work?

Reply:
The `check_method_call_by_name` call returns `Any` for all tuples, and creates the error message:

`error: "Tuple[]" has no attribute "__iter__" (not iterable)`.

The same is true for full tuples:

`error: "Tuple[int, str]" has no attribute "__iter__" (not iterable)`